internal/run/run.go

```diff
@@ -170,15 +171,16 @@ func (w *Workflow) RunWithVisualization(ctx context.Context) error {
 		)
 		logger.Println(msg)
 
-		if !w.hasGenerationAccess {
+		if w.generationAccess != nil && !w.generationAccess.AccessAllowed {
 			warningDate := time.Date(2024, time.March, 18, 0, 0, 0, 0, time.UTC)
 			daysToLimit := int(math.Round(warningDate.Sub(time.Now().Truncate(24*time.Hour)).Hours() / 24))
+			additionalLines := strings.Split(w.generationAccess.Message, "\n")
+			additionalLines = append(additionalLines, fmt.Sprintf("Please reach out to the Speakeasy team in the next %d days to ensure continued access.", daysToLimit))
```

Comment:
Strange that this same things is in two separate places?

Reply:
Ya good point will factor out.